v1/src/test/java/com/google/cloud/teleport/templates/TextToBigQueryStreamLT.java

```diff
@@ -0,0 +1,266 @@
+@TemplateLoadTest(TextToBigQueryStreaming.class)
+@RunWith(JUnit4.class)
+public class TextToBigQueryStreamLT extends TemplateLoadTestBase {
+  private static final String SPEC_PATH =
+      MoreObjects.firstNonNull(
+          TestProperties.specPath(), "gs://dataflow-templates/latest/Stream_GCS_Text_to_BigQuery");
+  private static final String ARTIFACT_BUCKET = TestProperties.artifactBucket();
+  private static final String TEST_ROOT_DIR =
+      TextToBigQueryStreamLT.class.getSimpleName().toLowerCase();
+  private static final String INPUT_PCOLLECTION =
+      "Read Text Data/Via ReadFiles/Read all via FileBasedSource/Read ranges.out0";
+  private static final String OUTPUT_PCOLLECTION =
+      "InsertIntoBigQuery/StreamingInserts/StreamingWriteTables/StreamingWrite/BatchedStreamingWrite.ViaBundleFinalization/ParMultiDo(BatchAndInsertElements).out1";
+  // schema should match schema supplied to generate fake records.
+  private static final Schema SCHEMA =
+      Schema.of(
+          Field.of("eventId", StandardSQLTypeName.STRING),
+          Field.of("eventTimestamp", StandardSQLTypeName.INT64),
+          Field.of("ipv4", StandardSQLTypeName.STRING),
+          Field.of("ipv6", StandardSQLTypeName.STRING),
+          Field.of("country", StandardSQLTypeName.STRING),
+          Field.of("username", StandardSQLTypeName.STRING),
+          Field.of("quest", StandardSQLTypeName.STRING),
+          Field.of("score", StandardSQLTypeName.INT64),
+          Field.of("completed", StandardSQLTypeName.BOOL));
+
+  // 35,000,000 messages of the given schema make up approximately 10GB
+  private static final int NUM_MESSAGES_FOR_10GB = 35000000;
+  private static final long TIMEOUT_FOR_10_GB_TEST_MINUTES = 30;
+  private static final long TIMEOUT_FOR_1_HOUR_TEST_MINUTES = 60;
+
+  private static ArtifactClient gcsClient;
+  private static BigQueryResourceManager bigQueryResourceManager;
+
+  @Before
+  public void setup() throws IOException {
+    bigQueryResourceManager =
+        BigQueryResourceManager.builder(testName, project).setCredentials(CREDENTIALS).build();
+
+    Storage storageClient = createStorageClient(CREDENTIALS);
+
+    gcsClient = GcsArtifactClient.builder(storageClient, ARTIFACT_BUCKET, TEST_ROOT_DIR).build();
+  }
+
+  @After
+  public void teardown() {
+    ResourceManagerUtils.cleanResources(bigQueryResourceManager, gcsClient);
+  }
+
+  @Test
+  public void testBacklog10gb() throws IOException, ParseException, InterruptedException {
+    testBacklog(Function.identity());
+  }
+
+  @Test
+  public void testSteadyState1hr() throws IOException, ParseException, InterruptedException {
+    testSteadyState1hr(Function.identity());
+  }
+
+  private void testBacklog(
+      Function<PipelineLauncher.LaunchConfig.Builder, PipelineLauncher.LaunchConfig.Builder>
+          paramsAdder)
+      throws IOException, ParseException, InterruptedException {
+    // upload schema files and save path
+    String jsonPath =
```

Comment:
We can upload the schema file and save path in the `setup` method since this is done for all the tests.